### PR TITLE
Add cu settings to 6.1.1

### PR DESCRIPTION
**Pull request by neon60**
add cu setting page

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -117,6 +117,7 @@ Our documentation is organized into the following categories:
   * [MI250](./conceptual/gpu-arch/mi250.md)
   * [MI300](./conceptual/gpu-arch/mi300.md)
 * [GPU memory](./conceptual/gpu-memory.md)
+* [Setting the number of CUs](./conceptual/setting-cus)
```
Comment (lpaoletti) [inline docs/index.md:120]: Adding this to Conceptual might not be correct - the title makes this look like procedure (with the gerund "Setting").
If this is a procedure, it should be in the How to section. 
If conceptual, it should be renamed. 

image

File: docs/conceptual/setting-cus.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    :keywords: AMD, ROCm, cu, number of cus
+
+.. _env-variables-reference:
+
+*************************************************************
+Setting the number of CUs
+*************************************************************
+
+When using GPUs to accelerate compute workloads, it sometimes becomes necessary
+to configure the hardware's usage of Compute Units (CU). This is a more advanced
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:12]: Perhaps we need to call this out as Tip.
If experimentation can cause bad things to happen like damaging the hardware, we should put in a warning.

File: docs/conceptual/setting-cus.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
+.. meta::
+    :description: Setting the number of CUs
+    :keywords: AMD, ROCm, cu, number of cus
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:3]:
```suggestion
    :keywords: CU, CUs, number of CUs, compute units
```

File: docs/conceptual/setting-cus.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+.. _env-variables-reference:
+
+*************************************************************
+Setting the number of CUs
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:8]:
```suggestion
Setting the number of compute units
```

File: docs/conceptual/setting-cus.rst (hunk 2 above)
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:12]:
```suggestion
to configure the hardware's usage of compute units (CU). This is a more advanced
```

File: docs/conceptual/setting-cus.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+When using GPUs to accelerate compute workloads, it sometimes becomes necessary
+to configure the hardware's usage of Compute Units (CU). This is a more advanced
+option, so please read this page before experimentation.
+
+The GPU driver provides two environment variables to set the number of CUs used. The
+first one is ``HSA_CU_MASK`` and the second one is ``ROC_GLOBAL_CU_MASK``. The main
+difference is that ``ROC_GLOBAL_CU_MASK`` sets the CU mask on queues created by the HIP
+or the OpenCL runtimes. While ``HSA_CU_MASK`` sets the mask on a lower level of queue
+creation in the driver, this mask will also be set for queues being profiled.
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:19]: Perhaps this instead:
The ROC_GLOBAL_CU_MASK variable sets the CU mask on queues created by HIP or OpenCL runtimes. The HSA_CU_MASK variable sets the mask on a lower level of queue creation in the driver. It also sets the mask on the queues being profiled.

File: docs/conceptual/setting-cus.rst [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    ID = [0-9][0-9]*                         ex. base 10 numbers
+    ID_list = (ID | ID-ID)[, (ID | ID-ID)]*  ex. 0,2-4,7
+    GPU_list = ID_list                       ex. 0,2-4,7
+    CU_list = 0x[0-F]* | ID_list             ex. 0x337F OR 0,2-4,7
+    CU_Set = GPU_list : CU_list              ex. 0,2-4,7:0-15,32-47 OR 0,2-4,7:0x337F
+    HSA_CU_MASK = CU_Set [; CU_Set]*         ex. 0,2-4,7:0-15,32-47; 3-9:0x337F
+
+The GPU indices are taken post ``ROCR_VISIBLE_DEVICES`` reordering. For GPUs listed,
+the listed or masked CUs will be enabled, the rest disabled. Unlisted GPUs will not
+be affected, their CUs will all be enabled.
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:34]:
```suggestion
The GPU indices are taken post ROCR_VISIBLE_DEVICES reordering. The listed or masked CUs are enabled for listed GPUs, and the others are disabled. Unlisted GPUs are not be affected, and their CUs are enabled.
```

File: docs/conceptual/setting-cus.rst [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+The parsing of the variable is stopped when a syntax error occurs. The erroneous set
+and the ones following will be ignored. Repeating GPU or CU IDs are a syntax error.
+Specifying a mask with no usable CUs (CU_list is 0x0) is a syntax error. For excluding
+GPU devices use ``ROCR_VISIBLE_DEVICES``.
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:39]:
```suggestion
The variable parsing stops when a syntax error occurs. The erroneous set and the following are ignored. Repeating GPU or CU IDs results in a syntax error. Specifying a mask with no usable CUs (CU_list is 0x0) results in a syntax error. To exclude GPU devices, use ``ROCR_VISIBLE_DEVICES``.
```

File: docs/conceptual/setting-cus.rst [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+These environment variables only affect ROCm software, not graphics applications.
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:41]: This should be a note.
These environment variables only affect ROCm software applications and not graphics applications.

File: docs/conceptual/setting-cus.rst [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+It's important to know that not all CU configurations are valid on all devices. For
+instance, on devices where two CUs can be combined into a WGP (for kernels running in
+WGP mode), it is not valid to disable only a single CU in a WGP. `This paper
+<https://www.cs.unc.edu/~otternes/papers/rtsj2022.pdf>`_ can provide more information
+about what to expect, when disabling CUs.
```
Comment (lpaoletti) [inline docs/conceptual/setting-cus.rst:47]:
```suggestion
Not all CU configurations are valid on all devices. For example, on devices where two CUs can be combined into a WGP (for kernels running in WGP mode), it’s not valid to disable only a single CU in a WGP. For more information about what to expect when disabling CUs, see the [Exploring AMD GPU Scheduling Details by Experimenting With “Worst Practices”](https://www.cs.unc.edu/~otternes/papers/rtsj2022.pdf) paper.

```

File: docs/index.md (hunk 1 above)
Comment (neon60) [reply docs/index.md:120]: Moved to how-to

File: docs/conceptual/setting-cus.rst (hunk 5 above)
Comment (neon60) [reply docs/conceptual/setting-cus.rst:21]: Fixed.

**Comment by neon60** [comment]
@lpaoletti
This page is already added to develop branch. After we accept the changes, I will apply the fixes on develop and 6.1.2 branches.

File: docs/conceptual/setting-cus.rst (hunk 2 above)
Comment (neon60) [reply docs/conceptual/setting-cus.rst]: Fixed.

File: docs/how-to/setting-cus.rst [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+The GPU indices are taken post ``ROCR_VISIBLE_DEVICES`` reordering. The listed or masked CUs are enabled for listed GPUs, and the others are disabled. Unlisted GPUs are not be affected, and their CUs are enabled.
+
+The variable parsing stops when a syntax error occurs. The erroneous set and the following are ignored. Repeating GPU or CU IDs results in a syntax error. Specifying a mask with no usable CUs (CU_list is 0x0) results in a syntax error. To exclude GPU devices, use ``ROCR_VISIBLE_DEVICES``.
+
+.. note::
+
+    These environment variables only affect ROCm software, not graphics applications.
+
+Not all CU configurations are valid on all devices. For example, on devices where two CUs can be combined into a WGP (for kernels running in WGP mode), it’s not valid to disable only a single CU in a WGP. For more information about what to expect when disabling CUs, see the [Exploring AMD GPU Scheduling Details by Experimenting With “Worst Practices”](https://www.cs.unc.edu/~otternes/papers/rtsj2022.pdf) paper.
```
Comment (lpaoletti) [inline docs/how-to/setting-cus.rst:41]: This hyperlink looks messed up
image

File: docs/how-to/setting-cus.rst [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    :description: Setting the number of CUs
+    :keywords: CU, CUs, number of CUs, compute units
+
+.. _env-variables-reference:
+
+*************************************************************
+Setting the number of compute units
+*************************************************************
+
+.. tip::
```
Comment (lpaoletti) [inline docs/how-to/setting-cus.rst:11]: Not good practice to start with a tip. 
Remove the tip box.

A better alternative is to move the tip box below the first two bullets.

File: docs/how-to/setting-cus.rst (hunk 10 above)
Comment (neon60) [reply docs/how-to/setting-cus.rst]: Fixed.

**Comment by neon60** [comment]
@lpaoletti Could you approve this? Is everything okay with this now?

**Review by lpaoletti — APPROVED** [review]

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**